rct229/rulesets/ashrae9012019/section19/section19rule18.py

```diff
@@ -0,0 +1,366 @@
+
+            if 9 <= MERV_rating <= 12:
+                MERV_adj = 0.5
+            elif MERV_rating > 23:
+                MERV_adj = 0.9
+            else:
+                MERV_adj = 0
+
+            A += MERV_adj * supply_flow_b / 4131
+            if any(
+                [
+                    baseline_system_type_compare(sys_type_b, target_sys_type, False)
+                    for target_sys_type in CONSTANT_VOLUME_SYS_TYPES
+                ]
+            ):
+                expected_BHP_b = ((0.00094 * supply_flow_b) + A).m * ureg("hp")
+                min_BHP_b = (0.00094 * supply_flow_b).m * ureg("hp")
+            elif any(
+                [
+                    baseline_system_type_compare(sys_type_b, target_sys_type, False)
+                    for target_sys_type in VARIABLE_VOLUME_SYS_TYPES
+                ]
+            ):
+                expected_BHP_b = ((0.0013 * supply_flow_b) + A).m * ureg("hp")
+                min_BHP_b = (0.0013 * supply_flow_b).m * ureg("hp")
+            else:
+                expected_BHP_b = ((0.00062 * supply_flow_b) + A).m * ureg("hp")
+                min_BHP_b = (0.00062 * supply_flow_b).m * ureg("hp")
+
+            expected_motor_efficiency_b = table_G3_9_1_lookup(expected_BHP_b)[
+                "nominal_full_load_efficiency"
+            ]
+            min_motor_efficiency_b = table_G3_9_1_lookup(min_BHP_b)[
+                "nominal_full_load_efficiency"
+            ]
+
+            expected_fan_wattage_b = (
+                expected_BHP_b * 746 * (1 / expected_motor_efficiency_b)
+            )
+            min_fan_wattage_b = min_BHP_b * 746 * (1 / min_motor_efficiency_b)
+
+            return {
+                "more_than_one_supply_fan_b": more_than_one_supply_fan_b,
+                "more_than_one_exhaust_fan_and_energy_rec_is_relevant_b": more_than_one_exhaust_fan_and_energy_rec_is_relevant_b,
+                "total_fan_power_b": total_fan_power_b,
+                "expected_fan_wattage_b": expected_fan_wattage_b,
+                "min_fan_wattage_b": min_fan_wattage_b,
+            }
+
+        def manual_check_required(self, context, calc_vals=None, data=None):
+            more_than_one_supply_fan_b = calc_vals["more_than_one_supply_fan_b"]
+            more_than_one_exhaust_fan_and_energy_rec_is_relevant_b = calc_vals[
+                "more_than_one_exhaust_fan_and_energy_rec_is_relevant_b"
+            ]
+            total_fan_power_b = calc_vals["total_fan_power_b"]
+            expected_fan_wattage_b = calc_vals["expected_fan_wattage_b"]
+
+            return not (
+                (
+                    not more_than_one_exhaust_fan_and_energy_rec_is_relevant_b
+                    and (
+                        (
+                            not more_than_one_supply_fan_b
+                            and std_equal(total_fan_power_b, expected_fan_wattage_b)
+                        )
+                        or (total_fan_power_b < expected_fan_wattage_b)
```

Comment:
We do not need to care about AHJ_RA_compare in this case. So case 2 and 3 are the same case that returns Failed.

Reply:
Good point! Addressed